pkg/utils/retry.go

```diff
@@ -0,0 +1,24 @@
+package utils
```

Comment:
I prefer not having utils packages, they end up being a dumpster-fire of unrelated code. 😂 

What do you think of calling this `retry` package?